### PR TITLE
174 list model

### LINKED ISSUE #174: List Model
**As a** product owner
**I want** to be able to store metadata associated with lists of chemicals
**So that** I can easily group those chemicals

A list can serve multiple purposes and that will be detailed out in functionality requests in the future, but for a MVP, we are just going to replicate what is in the current codebase.

A List Consists of the following fields:
Name = String (Less than 50 character, url safe, unique, required field)
Label = String (Less than 255 characters, unique, required field)
Short Description = String (Less than 1000 characters, required field)
Long Description = TEXT (required field)
list accessibility = accessibility type (issue #179) (required) 
Owners: (A list of users, default current user)
source_url: String (Less than 500 characters, optional field)
source_reference: String (Less than 500 characters, optional field)
source_doi: String (Less than 500 characters, optional field)
External Contact: Sub-object (optional)
    name
    email
    phone 
Date of source collection: (date, required)

**Acceptance Criteria**
A test to show when a substance history is created it works properly...

### PR DESCRIPTION
**Pull request by AScaduto**
Closes #174 

There is an issue that arose with this ticket. In the ticket description, the owners attribute is defined as (A list of users, default current user). The problem here is that Django has issues with providing a default in a many to many relationship. 

More info regarding that can be found below:

https://code.djangoproject.com/ticket/2750

The way that we would go about obtaining the current user pk here would be by setting the default to  
# default=get_current_user_pk 

However based on the issue above, there is no valid way to test if this is working as Django does not allow you to properly authenticate from the command line and when creating a list object, I am continuously receiving an empty set of data.

### CODE REVIEW

**Comment by AScaduto** [comment]
Another thing that I wanted to clarify with this model is that the descriptions for label and short_description state that they both have different character limits than their ControlledVocabulary counterparts. I went ahead and overrode them and created new ones in the List model but I was not sure if this was intended.

File: chemreg/lists/models.py [hunk 1]
```diff
@@ -25,3 +28,50 @@ class IdentifierType(ControlledVocabulary):
...
+        email (str): (Less than 49 characters, unique, required)
+        phone (str): (Less than 15 characters, unique, required)
+    """
+
+    name = models.CharField(max_length=49, unique=True, blank=False)
+    email = models.CharField(max_length=49, unique=True, blank=False)
+    phone = models.CharField(max_length=15, unique=True, blank=False)
+
+
+class List(ControlledVocabulary):
```
Comment (michael-barbour) [inline chemreg/lists/models.py:47]: Can you double check with Chris that this is supposed to be a controlled vocabulary?  His original model request does not ask for the "Depricated" field.

Comment (AScaduto) [reply chemreg/lists/models.py:47]: 👍 See my comment above

**Comment by cmgrulke** [comment]
So I don't really consider a list a controlled vocabulary (though if that is how you decided to implement it, I can understand why since it has many of the same fields).  List objects are likely going to become pretty complex and have many complex business processes associated with them whereas controlled vocabularies are generally just boring objects that are simple annotations for other important business relevant classes.  

The delete/deprecation behavior of lists should definitely not be the same as that for a controlled vocabulary, but will be specified in later tickets.

**Pull request merged**